docs/docs/introducing-jsx.md

```diff
@@ -0,0 +1,202 @@
+
+This funny tag syntax is neither a string nor HTML.
+
+It is called JSX, and it is a syntax extension to JavaScript. We recommend using it with React to describe what the UI should look like. JSX may remind you of a template language, but it comes with a full power of JavaScript.
+
+>**Note:**
+>
+>Many people get turned off by JSX the first time they see it. One might think it violates separation of concerns but we believe [otherwise](https://www.youtube.com/watch?v=x7cQ3mrcKaY).
+>
+>We encourage you to [give it five minutes](https://signalvnoise.com/posts/3124-give-it-five-minutes). Facebook and many other companies successfully use it in production both for simple and complex apps.
+>
+>Also note that JSX is optional, and you can [use React without JSX](/react/docs/react-without-jsx.html).
+
+### Embedding Expressions in JSX
+
+JSX comes with a full power of JavaScript.
+
+You can embed any [JavaScript expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions) in JSX by wrapping it in the curly braces.
+
+For example, `2 + 2`, `user.name`, and `formatName(user)` are all valid expressions:
+
+```js{12}
+function formatName(user) {
+  return user.firstName + ' ' + user.lastName;
+}
+
+let user = {
+  firstName: 'Harper',
+  lastName: 'Perez'
+};
+
+let element = (
+  <h1>
+    Hello, {formatName(user)}!
+  </h1>
+);
+```
+
+**[Try it on Codepen.](http://codepen.io/gaearon/pen/PGEjdG?editors=0010)**
+
+Note that we wrapped JSX in parens and split it over multiple lines for readability.
+
+>**Caveat:**
+>
+>You can't use `if` or `for` statements *inside* of JSX because they are not [expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions).
+
+### JSX is an Expression Too
+
+After compilation, JSX expressions become regular JavaScript objects.
+
+This means that you can use JSX inside of `if` statements and `for` loops, assign it to variables, accept it as arguments, and return it from functions:
+
+```js{3,5}
+function getGreeting(user) {
+  if (user) {
+    return <h1>Hello, {formatName(user.name)}!</h1>;
+  } else {
+    return <h1>Hello, Stranger.</h1>;
+  }
+}
+```
+
+### Specifying Attributes with JSX
+
+You may use quotes to specify string literals as attributes:
+
+```js
+let element = <div tabIndex="0"></div>;
+```
+
+You may also use the curly braces to embed a JavaScript expression in an attribute:
+
+```js
+let element = <img src={user.avatarUrl}></img>;
+```
+
+### Specifying Children with JSX
+
+If a tag is empty, you may close it immediately with `/>`, like in XML:
+
+```js
+let element = <img src={user.avatarUrl} />;
+```
+
+JSX tags may contain children:
+
+```js
+let element = (
+  <div>
+    <h1>Hello!</h1>
+    <h2>Good to see you here.</h2>
+  </div>
+);
+```
+
+>**Caveat:**
+>
+>Since JSX is closer to JavaScript than HTML, it uses property names as defined in the DOM specification rather than HTML attribute names.
```

Comment:
Let's be clear that this is React DOM that makes this choice, not JSX or React. Maybe also link to the attribute reference page.

nit: It isn't true that we use the DOM names in all cases; sometimes we camelcase them where the DOM prop is lowercase.